hls4ml/model/graph.py

```diff
@@ -506,6 +506,8 @@ def insert_node(self, node, before=None, input_idx=0):
 
         if next_node is not None:
             next_node.inputs[input_idx] = node.outputs[0]
+        else:
+            self.outputs = [node.outputs[0] if name == prev_node.outputs[0] else name for name in self.outputs]
```

Comment:
Is it obvious that prev_node.outputs[0] needs to be replaced with node.outputs[0] node is added? Both seem to have some rationale in my mind.

Reply:
The original logic assumes all outputs not consumed by any other node is the model output, and vice versa. This weakens the assumption, in which when a node is appended to an output node, it assumed to add a stage in the same pipeline but not causing other side effects. Otherwise, the appended node stays stray after operation, if not otherwise fixed by an optimizer. Tests will fail if this is not included.

Reply:
I didn't quite follow. Isn't this to decide when you insert a node after a node whose output is in the outputs, whether that stays in the outputs or the new node output is in the outputs? But really I don't know which is better, so if you say this works better, I am fine with it.